### PR TITLE
Initial Enchantment Table Implementation

### PR DESCRIPTION
**Pull request by aPunch**
This pull adds enchantment table controllers and cleans up some code in existing controllers and blocks.

I had some trouble with merging here, so there are two commits. Trying to rebase made it worse, so hopefully you'll let the one-commit-only rule slide this time. :P

### CODE REVIEW

File: src/main/java/org/spout/vanilla/controller/block/DispenserController.java [hunk 1]
```diff
@@ -32,10 +32,11 @@
 import org.spout.vanilla.material.VanillaMaterials;
 
 public class DispenserController extends VanillaBlockController {
-	private final DispenserInventory inventory = new DispenserInventory(this);
+	private final DispenserInventory inventory;
 
 	public DispenserController() {
 		super(VanillaControllerTypes.DISPENSER, VanillaMaterials.DISPENSER);
+		inventory = new DispenserInventory(this);
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/controller/block/DispenserController.java:39]: Why is this necessary?

File: src/main/java/org/spout/vanilla/controller/block/FurnaceController.java [hunk 2]
```diff
@@ -136,7 +138,8 @@ public void onTick(float dt) {
 					if (c instanceof VanillaPlayer) {
 						VanillaPlayer controller = (VanillaPlayer) c;
 						int window = controller.getWindowId();
-						sendPacket(controller.getPlayer(), new ProgressBarMessage(window, Furnace.FIRE_ICON, (int) burnIncrement), new ProgressBarMessage(window, Furnace.PROGRESS_ARROW, (int) progressIncrement));
+						sendPacket(controller.getPlayer(), new ProgressBarMessage(window, Furnace.FIRE_ICON, (int) burnIncrement), new ProgressBarMessage(window,
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/controller/block/FurnaceController.java:141]: No line breaks please :)

File: src/main/java/org/spout/vanilla/controller/world/PointObserver.java [hunk 3]
```diff
@@ -65,9 +65,4 @@ public void onAttached() {
 		getParent().setViewDistance(256);
 		currPoint = getParent().getPosition();
 	}
-	
-	@Override
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/controller/world/PointObserver.java:69]: Why is this being removed?

File: src/main/java/org/spout/vanilla/material/block/misc/Chest.java [hunk 4]
```diff
@@ -64,18 +61,24 @@ public void loadProperties() {
...
 		setController(VanillaControllerTypes.CHEST);
 	}
-	
+
 	public boolean isDouble(Block block) {
 		Point pos = block.getPosition();
 		World world = pos.getWorld();
 		VanillaBlockMaterial chest = VanillaMaterials.CHEST;
-		if (world.getBlock(pos.add(1, 0, 0)).getMaterial() == chest || world.getBlock(pos.subtract(1, 0, 0)).getMaterial() == chest || world.getBlock(pos.add(0, 0, 1)).getMaterial() == chest || world.getBlock(pos.subtract(0, 0, 1)).getMaterial() == chest) {
+		if (world.getBlock(pos.add(1, 0, 0)).getMaterial() == chest || world.getBlock(pos.subtract(1, 0, 0)).getMaterial() == chest
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/material/block/misc/Chest.java:69]: Again, no line breaks :p

**Comment by windy1** [comment]
Also, your commits need to be squashed into one :)

**Comment by Zidane** [comment]
Forget my message...didn't see you actually did sign it off. I still do need you to rebase as you will need to fix what @W1ndwaker mentioned.

File: src/main/java/org/spout/vanilla/controller/block/DispenserController.java (hunk 1 above)
Comment (aPunch) [inline src/main/java/org/spout/vanilla/controller/block/DispenserController.java:39]: Not sure what you mean...I didn't originally implement that variable. I just initialized it in the constructor because I'm trying to cleanup and standardize some code.

File: src/main/java/org/spout/vanilla/controller/world/PointObserver.java (hunk 3 above)
Comment (aPunch) [inline src/main/java/org/spout/vanilla/controller/world/PointObserver.java:69]: This was a merge fail.

File: src/main/java/org/spout/vanilla/material/block/misc/Chest.java (hunk 4 above)
Comment (aPunch) [inline src/main/java/org/spout/vanilla/material/block/misc/Chest.java:69]: I'm not opposed to standardized formatting, but I find it odd that line breaks are enforced and properly spaced comments are not! :P

// This is a good comment.
//This is a bad comment. :(

**Comment by aPunch** [comment]
I'm going to close this and open a new PR. I've made a few mistakes that have piled up and it has turned into a git nightmare.
